src/storage-operators/src/persist_source.rs

```diff
@@ -210,6 +222,8 @@ pub fn persist_source_core<'g, G>(
     until: Antichain<Timestamp>,
     map_filter_project: Option<&mut MfpPlan>,
     flow_control: Option<FlowControl<RefinedScope<'g, G>>>,
+    // If Some, an override for the default listen sleep retry parameters.
+    listen_sleep: Option<impl Fn() -> RetryParameters + 'static>,
```

Comment:
I find it a bit weird that this is used to thread down one of the two retry policies from the Persist config, but the listen polling it gets threaded all the way down for already has access to that config. So the amount of extra "information" that this parameter threads through is, in practice, a boolean.

If it were me I'd probably just thread the boolean down. (Or, better, a custom enum.) But I don't think it's a blocker for the PR.

Reply:
Thought about this overnight. I don't love the boolean, but I'd be okay with the enum. However, I think I still prefer what I have, here's why:

persist-txn is meant to be an abstraction entirely on top of persist (we're adding some persist-txn motivated performance and ergonomics things, but for the most part this remains true). My mental model of this override is persist exposing an API of "okay if you really know what you're doing go ahead..." to persist-txn. Unfortunately, this is quite muddled by the LD plumbing for the override _also_ living in persist-client, but I view this as convenience so that I don't have to make a PersistTxnParameters and such. An enum would lean more into persist-client concerning itself with persist-txn things and I'd rather instead do a followup at some point (once the flags stuff I showed you is ready) to just pull the configs for this out of persist-client and into persist-txn, at which point it all makes more sense again.